### PR TITLE
Remove unused context parameter

### PR DESCRIPTION
**Pull request by justinmc**
I noticed a `context` parameter that was unused in these methods, so I've removed them.  This could be a breaking change, but because it was just introduced, it's probably unlikely.

### CODE REVIEW

File: packages/flutter/test/material/spell_check_suggestions_toolbar_test.dart [hunk 1]
```diff
@@ -85,4 +86,29 @@ void main() {
     final double toolbarY = tester.getTopLeft(findSpellCheckSuggestionsToolbar()).dy;
     expect(toolbarY, equals(expectedToolbarY));
   });
+
+  test('buildButtonItems builds only a delete button when no suggestions', () {
```
Comment (camsim99) [inline packages/flutter/test/material/spell_check_suggestions_toolbar_test.dart:90]: Great addition!

File: packages/flutter/test/cupertino/spell_check_suggestions_toolbar_test.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/services.dart';
+
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  testWidgets('buildButtonItems builds a "No Replacements Found" button when no suggestions', (WidgetTester tester) async {
```
Comment (camsim99) [inline packages/flutter/test/cupertino/spell_check_suggestions_toolbar_test.dart:13]: Great addition!!

**Review by camsim99 — APPROVED** [review]
This seems legitimate to me. Thanks!

**Comment by justinmc** [comment]
I agree, but in this case it's a public API, and we could potentially be providing the parameter to Flutter users even though it's not used internally in the framework itself.

I'm not sure if we have a lint that would catch it for a private API.  We should.

**Comment by Hixie** [comment]
This is a surprising change to me. Generally all build methods in public APIs should have the `context`, otherwise they can't do things like `MediaQuery.of` or `CupertinoTheme.of` or whatnot. Is this not that kind of build method?

**Comment by justinmc** [comment]
Hmm I'll look at this more closely tomorrow, maybe this is just a brain fart on my part as I tried to clean this up.  I think I didn't think of it as a typical builder because it doesn't return a Widget.  Also there is editableTextState.context, but maybe that's not what should be used here.

**Comment by Hixie** [comment]
(to be clear i'm not familiar with this code and am quite ready to believe these aren't normal builders and that it's fine to make this change)

**Comment by justinmc** [comment]
Ok after taking a closer look, I'm going to advocate for keeping it as-is (no `context` parameter).  These `buildButtonItems` methods are "building" information about the buttons, which are later turned into widgets elsewhere.  They shouldn't need to look up things like MediaQuery or Theme.  The one reason to use `context` would be for localizations, but I think it makes sense to get that from `editableTextState`.

Maybe the `build` name was a misnomer here.  I think these methods might be more analogous to [getEditableButtonItems](https://github.com/flutter/flutter/blob/8784eb1d8e30f2000b2c393f156e41257ee885e6/packages/flutter/lib/src/widgets/editable_text.dart#L1832) than a builder.

I'm definitely open to be proven wrong here though.  I'll keep an eye out if anyone opens an issue about this or shows confusion for lack of a BuildContext parameter.  Also, @camsim99 you're the original author here, let me know if you think I'm not on track with what you intended for these.

**Pull request merged**